Synthesis failed for google-oauth-java-client

Hello! Autosynth couldn't regenerate google-oauth-java-client. :broken_heart:

Here's the output from running `synth.py`:

```
 ([#456](https://www.github.com/googleapis/google-oauth-java-client/issues/456)) ([fc75233](https://www.github.com/googleapis/google-oauth-java-client/commit/fc752336af9cbdb9b2ed816a63d7bd3d8d1e2778))
---


This PR was generated with [Release Please](https://github.com/googleapis/release-please).

Source-Author: release-please[bot] <55107282+release-please[bot]@users.noreply.github.com>
Source-Date: Mon Jun 29 20:52:05 2020 +0000
Source-Repo: googleapis/google-oauth-java-client
Source-Sha: fa8c6ba33d9066ce5b14aa59f9a2c9972a8e95aa
Source-Link: https://github.com/googleapis/google-oauth-java-client/commit/fa8c6ba33d9066ce5b14aa59f9a2c9972a8e95aa
[autosynth-self-0 177ecad] chore: release 1.31.0
 5 files changed, 5 insertions(+), 5 deletions(-)
2020-06-29 14:34:40,128 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 177ecad chore: release 1.31.0
2020-06-29 14:34:40,136 autosynth [DEBUG] > Running: git checkout autosynth-self
Switched to branch 'autosynth-self'
2020-06-29 14:34:40,144 autosynth [DEBUG] > Running: git diff HEAD..autosynth-self-0 -- . :(exclude)synth.metadata
2020-06-29 14:34:40,150 autosynth [DEBUG] > Running: git diff HEAD autosynth-self-0
2020-06-29 14:34:40,155 autosynth [DEBUG] > Running: git apply /tmpfs/tmp/tmpf6_itl6x/autosynth-self-0.patch
2020-06-29 14:34:40,160 autosynth [DEBUG] > Running: git add -A
2020-06-29 14:34:40,167 autosynth [DEBUG] > Running: git commit -m changes without context

        autosynth cannot find the source of changes triggered by earlier changes in this
        repository, or by version upgrades to tools such as linters.
[autosynth-self c3f3c8f] changes without context
 5 files changed, 5 insertions(+), 5 deletions(-)
2020-06-29 14:34:40,177 autosynth [DEBUG] > Running: git diff HEAD..autosynth-self-1 -- . :(exclude)synth.metadata
2020-06-29 14:34:40,182 autosynth [DEBUG] > Running: git diff HEAD..autosynth-self-1 -- synth.metadata
2020-06-29 14:34:40,187 autosynth [DEBUG] > Running: git push --force origin autosynth-self
To https://github.com/googleapis/google-oauth-java-client.git
 + c646d29...c3f3c8f autosynth-self -> autosynth-self (forced update)
2020-06-29 14:34:43,983 autosynth [DEBUG] > Running: git log -1 --pretty=%b
2020-06-29 14:34:46,928 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-29 14:34:46,936 autosynth [DEBUG] > Running: git checkout fa8c6ba33d9066ce5b14aa59f9a2c9972a8e95aa
Note: checking out 'fa8c6ba33d9066ce5b14aa59f9a2c9972a8e95aa'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at fa8c6ba chore: release 1.31.0 (#473)
2020-06-29 14:34:46,944 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Previous HEAD position was 682c0c3 build(nodejs): update lint ignore rules (#488)
HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-06-29 14:34:46,966 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-06-29 14:34:46,971 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-06-29 14:34:46,978 autosynth [INFO] > Running synthtool
2020-06-29 14:34:46,979 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-29 14:34:46,979 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-06-29 14:34:46,982 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-29 14:34:47,244 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-06-29 14:34:47,407 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-29 14:34:47,434 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-29 14:34:47,499 autosynth [ERROR] > Synthesis failed
2020-06-29 14:34:47,500 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at fa8c6ba chore: release 1.31.0 (#473)
2020-06-29 14:34:47,507 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-29 14:34:47,514 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-29 14:34:47,723 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-06-29 14:34:47,723 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/2e2af012-7d46-45a9-9790-2acbfe2ab896/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
6b756a9532bb
2020-06-30 01:34:23,058 autosynth [DEBUG] > Running: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-06-30 01:34:23,061 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-06-30 01:34:23,064 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-06-30 01:34:23,067 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-06-30 01:34:23,071 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-06-30 01:34:23,074 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-06-30 01:34:23,077 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-06-30 01:34:23,081 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-06-30 01:34:23,085 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-06-30 01:34:23,088 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-06-30 01:34:23,092 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-06-30 01:34:23,095 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-06-30 01:34:23,099 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-06-30 01:34:23,103 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-06-30 01:34:23,107 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-06-30 01:34:23,377 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-06-30 01:34:23,522 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-30 01:34:23,532 autosynth [DEBUG] > Running: git checkout fa8c6ba33d9066ce5b14aa59f9a2c9972a8e95aa
Note: checking out 'fa8c6ba33d9066ce5b14aa59f9a2c9972a8e95aa'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at fa8c6ba chore: release 1.31.0 (#473)
2020-06-30 01:34:23,538 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-06-30 01:34:23,543 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-06-30 01:34:23,547 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-06-30 01:34:23,552 autosynth [INFO] > Running synthtool
2020-06-30 01:34:23,553 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-30 01:34:23,553 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-06-30 01:34:23,555 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-30 01:34:23,779 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-06-30 01:34:23,919 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-30 01:34:23,943 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-30 01:34:23,989 autosynth [ERROR] > Synthesis failed
2020-06-30 01:34:23,989 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at fa8c6ba chore: release 1.31.0 (#473)
2020-06-30 01:34:23,995 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-30 01:34:24,001 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-30 01:34:24,205 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-06-30 01:34:24,206 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/af3f101c-95bc-4779-bb24-a1b6ea1842ea/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
06-30 14:36:44,450 autosynth [DEBUG] > Running: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-06-30 14:36:44,453 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-06-30 14:36:44,456 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-06-30 14:36:44,459 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-06-30 14:36:44,462 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-06-30 14:36:44,465 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-06-30 14:36:44,468 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-06-30 14:36:44,471 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-06-30 14:36:44,474 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-06-30 14:36:44,477 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-06-30 14:36:44,480 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-06-30 14:36:44,483 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-06-30 14:36:44,486 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-06-30 14:36:44,490 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-06-30 14:36:44,494 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-06-30 14:36:44,854 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-06-30 14:36:44,976 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-30 14:36:44,987 autosynth [DEBUG] > Running: git checkout e97b2ce3cad2e6e67952bb164fee1d07f5655a15
Note: checking out 'e97b2ce3cad2e6e67952bb164fee1d07f5655a15'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at e97b2ce chore: release 1.31.1-SNAPSHOT (#484)
2020-06-30 14:36:44,992 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-06-30 14:36:44,998 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-06-30 14:36:45,001 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-06-30 14:36:45,006 autosynth [INFO] > Running synthtool
2020-06-30 14:36:45,007 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-30 14:36:45,007 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-06-30 14:36:45,009 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-30 14:36:45,230 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-06-30 14:36:45,376 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-30 14:36:45,400 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-30 14:36:45,445 autosynth [ERROR] > Synthesis failed
2020-06-30 14:36:45,446 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at e97b2ce chore: release 1.31.1-SNAPSHOT (#484)
2020-06-30 14:36:45,451 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-30 14:36:45,456 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-30 14:36:45,634 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-06-30 14:36:45,634 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/064d885f-e246-4a73-93ae-61c4d27d9d17/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
07-01 01:34:35,773 autosynth [DEBUG] > Running: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-01 01:34:35,778 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-01 01:34:35,782 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-01 01:34:35,786 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-01 01:34:35,790 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-01 01:34:35,795 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-01 01:34:35,800 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-01 01:34:35,804 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-01 01:34:35,809 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-01 01:34:35,813 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-01 01:34:35,817 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-01 01:34:35,821 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-01 01:34:35,825 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-01 01:34:35,830 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-01 01:34:35,835 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-01 01:34:36,218 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-01 01:34:36,365 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-01 01:34:36,376 autosynth [DEBUG] > Running: git checkout e97b2ce3cad2e6e67952bb164fee1d07f5655a15
Note: checking out 'e97b2ce3cad2e6e67952bb164fee1d07f5655a15'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at e97b2ce chore: release 1.31.1-SNAPSHOT (#484)
2020-07-01 01:34:36,381 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-01 01:34:36,388 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-01 01:34:36,392 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-01 01:34:36,397 autosynth [INFO] > Running synthtool
2020-07-01 01:34:36,398 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-01 01:34:36,398 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-01 01:34:36,400 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-01 01:34:36,610 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-01 01:34:36,746 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-01 01:34:36,768 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-01 01:34:36,814 autosynth [ERROR] > Synthesis failed
2020-07-01 01:34:36,814 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at e97b2ce chore: release 1.31.1-SNAPSHOT (#484)
2020-07-01 01:34:36,820 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-01 01:34:36,826 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-01 01:34:37,005 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-01 01:34:37,005 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/b53d2c2a-fda1-4a85-a3d8-dba91b663889/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-01 14:34:20,886 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-01 14:34:20,890 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-01 14:34:20,893 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-01 14:34:20,897 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-01 14:34:20,901 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-01 14:34:20,904 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-01 14:34:20,908 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-01 14:34:20,911 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-01 14:34:20,914 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-01 14:34:20,918 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-01 14:34:20,921 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-01 14:34:20,925 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-01 14:34:20,929 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-01 14:34:20,933 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-01 14:34:21,318 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-01 14:34:21,449 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-01 14:34:21,460 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-01 14:34:21,467 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-01 14:34:21,474 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-01 14:34:21,478 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-01 14:34:21,484 autosynth [INFO] > Running synthtool
2020-07-01 14:34:21,485 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-01 14:34:21,485 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-01 14:34:21,487 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-01 14:34:21,704 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-01 14:34:21,846 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-01 14:34:21,870 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-01 14:34:21,918 autosynth [ERROR] > Synthesis failed
2020-07-01 14:34:21,919 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-01 14:34:21,927 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-01 14:34:21,933 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-01 14:34:22,130 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-01 14:34:22,130 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/8f9b2ee6-9eab-4b87-867c-98d5f8fc5313/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-02 01:34:34,295 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-02 01:34:34,298 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-02 01:34:34,301 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-02 01:34:34,304 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-02 01:34:34,308 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-02 01:34:34,311 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-02 01:34:34,314 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-02 01:34:34,317 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-02 01:34:34,320 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-02 01:34:34,323 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-02 01:34:34,327 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-02 01:34:34,330 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-02 01:34:34,333 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-02 01:34:34,337 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-02 01:34:34,606 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-02 01:34:34,771 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-02 01:34:34,783 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-02 01:34:34,789 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-02 01:34:34,795 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-02 01:34:34,798 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-02 01:34:34,804 autosynth [INFO] > Running synthtool
2020-07-02 01:34:34,804 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-02 01:34:34,804 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-02 01:34:34,807 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-02 01:34:35,034 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-02 01:34:35,174 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-02 01:34:35,197 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-02 01:34:35,237 autosynth [ERROR] > Synthesis failed
2020-07-02 01:34:35,238 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-02 01:34:35,244 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-02 01:34:35,249 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-02 01:34:35,439 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-02 01:34:35,439 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/4ca920bc-93a2-47e1-82cb-e2726d376e7a/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-02 14:34:15,580 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-02 14:34:15,583 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-02 14:34:15,586 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-02 14:34:15,589 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-02 14:34:15,592 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-02 14:34:15,595 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-02 14:34:15,598 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-02 14:34:15,601 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-02 14:34:15,604 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-02 14:34:15,607 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-02 14:34:15,610 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-02 14:34:15,613 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-02 14:34:15,617 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-02 14:34:15,620 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-02 14:34:16,029 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-02 14:34:16,191 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-02 14:34:16,201 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-02 14:34:16,206 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-02 14:34:16,211 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-02 14:34:16,215 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-02 14:34:16,219 autosynth [INFO] > Running synthtool
2020-07-02 14:34:16,220 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-02 14:34:16,220 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-02 14:34:16,222 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-02 14:34:16,436 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-02 14:34:16,570 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-02 14:34:16,592 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-02 14:34:16,629 autosynth [ERROR] > Synthesis failed
2020-07-02 14:34:16,630 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-02 14:34:16,637 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-02 14:34:16,642 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-02 14:34:16,866 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-02 14:34:16,866 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/924dc30a-4d68-41cb-9955-faced2a5ea49/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-03 01:34:25,489 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-03 01:34:25,492 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-03 01:34:25,495 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-03 01:34:25,498 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-03 01:34:25,501 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-03 01:34:25,503 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-03 01:34:25,506 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-03 01:34:25,509 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-03 01:34:25,512 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-03 01:34:25,515 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-03 01:34:25,518 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-03 01:34:25,521 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-03 01:34:25,524 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-03 01:34:25,528 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-03 01:34:25,769 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-03 01:34:25,909 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-03 01:34:25,919 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-03 01:34:25,924 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-03 01:34:25,928 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-03 01:34:25,931 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-03 01:34:25,936 autosynth [INFO] > Running synthtool
2020-07-03 01:34:25,936 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-03 01:34:25,936 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-03 01:34:25,938 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-03 01:34:26,147 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-03 01:34:26,277 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-03 01:34:26,299 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-03 01:34:26,341 autosynth [ERROR] > Synthesis failed
2020-07-03 01:34:26,341 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-03 01:34:26,347 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-03 01:34:26,353 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-03 01:34:26,621 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-03 01:34:26,621 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/fb9759e4-26ce-4470-b984-f37c1e6ef1a2/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-03 14:34:09,108 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-03 14:34:09,111 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-03 14:34:09,114 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-03 14:34:09,117 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-03 14:34:09,120 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-03 14:34:09,123 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-03 14:34:09,125 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-03 14:34:09,128 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-03 14:34:09,131 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-03 14:34:09,134 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-03 14:34:09,137 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-03 14:34:09,140 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-03 14:34:09,143 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-03 14:34:09,147 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-03 14:34:09,428 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-03 14:34:09,587 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-03 14:34:09,597 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-03 14:34:09,601 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-03 14:34:09,606 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-03 14:34:09,609 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-03 14:34:09,613 autosynth [INFO] > Running synthtool
2020-07-03 14:34:09,614 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-03 14:34:09,614 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-03 14:34:09,615 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-03 14:34:09,829 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-03 14:34:09,967 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-03 14:34:09,989 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-03 14:34:10,025 autosynth [ERROR] > Synthesis failed
2020-07-03 14:34:10,026 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-03 14:34:10,031 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-03 14:34:10,036 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-03 14:34:10,258 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-03 14:34:10,258 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/97c9da19-04fb-480c-8fac-c72d90a592a7/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-04 01:34:14,346 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-04 01:34:14,349 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-04 01:34:14,352 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-04 01:34:14,355 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-04 01:34:14,358 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-04 01:34:14,361 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-04 01:34:14,365 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-04 01:34:14,368 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-04 01:34:14,371 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-04 01:34:14,374 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-04 01:34:14,377 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-04 01:34:14,380 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-04 01:34:14,384 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-04 01:34:14,388 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-04 01:34:14,640 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-04 01:34:14,774 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-04 01:34:14,785 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-04 01:34:14,790 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-04 01:34:14,796 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-04 01:34:14,799 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-04 01:34:14,804 autosynth [INFO] > Running synthtool
2020-07-04 01:34:14,805 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-04 01:34:14,805 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-04 01:34:14,807 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-04 01:34:15,027 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-04 01:34:15,165 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-04 01:34:15,189 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-04 01:34:15,232 autosynth [ERROR] > Synthesis failed
2020-07-04 01:34:15,233 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-04 01:34:15,238 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-04 01:34:15,243 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-04 01:34:15,442 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-04 01:34:15,442 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/0c8b112d-b2d4-4c18-aeee-2e028e08cd8c/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-04 14:34:03,622 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-04 14:34:03,625 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-04 14:34:03,628 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-04 14:34:03,631 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-04 14:34:03,634 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-04 14:34:03,637 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-04 14:34:03,640 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-04 14:34:03,643 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-04 14:34:03,646 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-04 14:34:03,649 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-04 14:34:03,651 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-04 14:34:03,654 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-04 14:34:03,658 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-04 14:34:03,662 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-04 14:34:03,918 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-04 14:34:04,076 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-04 14:34:04,087 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-04 14:34:04,091 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-04 14:34:04,096 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-04 14:34:04,099 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-04 14:34:04,104 autosynth [INFO] > Running synthtool
2020-07-04 14:34:04,104 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-04 14:34:04,104 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-04 14:34:04,106 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-04 14:34:04,306 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-04 14:34:04,432 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-04 14:34:04,453 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-04 14:34:04,488 autosynth [ERROR] > Synthesis failed
2020-07-04 14:34:04,488 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-04 14:34:04,494 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-04 14:34:04,499 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-04 14:34:04,689 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-04 14:34:04,689 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/222cbcc1-e4bd-4cd8-9e78-11b9f1e97cf9/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-05 01:34:10,880 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-05 01:34:10,884 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-05 01:34:10,887 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-05 01:34:10,890 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-05 01:34:10,893 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-05 01:34:10,896 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-05 01:34:10,899 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-05 01:34:10,902 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-05 01:34:10,905 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-05 01:34:10,908 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-05 01:34:10,912 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-05 01:34:10,914 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-05 01:34:10,918 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-05 01:34:10,922 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-05 01:34:11,170 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-05 01:34:11,315 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-05 01:34:11,326 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-05 01:34:11,331 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-05 01:34:11,337 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-05 01:34:11,341 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-05 01:34:11,346 autosynth [INFO] > Running synthtool
2020-07-05 01:34:11,346 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-05 01:34:11,346 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-05 01:34:11,348 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-05 01:34:11,555 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-05 01:34:11,687 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-05 01:34:11,709 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-05 01:34:11,752 autosynth [ERROR] > Synthesis failed
2020-07-05 01:34:11,752 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-05 01:34:11,757 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-05 01:34:11,763 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-05 01:34:11,966 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-05 01:34:11,966 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/b1a991ee-cb9e-4794-9390-01f6f53610dd/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-05 14:34:17,270 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-05 14:34:17,274 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-05 14:34:17,277 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-05 14:34:17,280 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-05 14:34:17,283 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-05 14:34:17,286 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-05 14:34:17,289 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-05 14:34:17,292 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-05 14:34:17,295 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-05 14:34:17,298 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-05 14:34:17,302 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-05 14:34:17,305 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-05 14:34:17,309 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-05 14:34:17,313 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-05 14:34:17,697 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-05 14:34:17,823 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-05 14:34:17,835 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-05 14:34:17,841 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-05 14:34:17,847 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-05 14:34:17,850 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-05 14:34:17,856 autosynth [INFO] > Running synthtool
2020-07-05 14:34:17,856 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-05 14:34:17,856 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-05 14:34:17,859 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-05 14:34:18,072 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-05 14:34:18,210 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-05 14:34:18,233 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-05 14:34:18,281 autosynth [ERROR] > Synthesis failed
2020-07-05 14:34:18,282 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-05 14:34:18,288 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-05 14:34:18,294 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-05 14:34:18,477 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-05 14:34:18,477 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/47f4d271-f727-41fb-b026-f169657855e2/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-06 01:34:08,553 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-06 01:34:08,556 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-06 01:34:08,559 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-06 01:34:08,562 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-06 01:34:08,565 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-06 01:34:08,568 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-06 01:34:08,572 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-06 01:34:08,575 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-06 01:34:08,578 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-06 01:34:08,581 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-06 01:34:08,584 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-06 01:34:08,587 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-06 01:34:08,591 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-06 01:34:08,594 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-06 01:34:09,019 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-06 01:34:09,173 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-06 01:34:09,183 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-06 01:34:09,189 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-06 01:34:09,195 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-06 01:34:09,198 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-06 01:34:09,203 autosynth [INFO] > Running synthtool
2020-07-06 01:34:09,203 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-06 01:34:09,203 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-06 01:34:09,205 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-06 01:34:09,411 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-06 01:34:09,541 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-06 01:34:09,562 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-06 01:34:09,600 autosynth [ERROR] > Synthesis failed
2020-07-06 01:34:09,601 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-06 01:34:09,607 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-06 01:34:09,612 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-06 01:34:09,806 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-06 01:34:09,806 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/0dd008b8-1daf-4862-a402-59299db3437b/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-06 14:34:11,324 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-06 14:34:11,327 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-06 14:34:11,330 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-06 14:34:11,333 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-06 14:34:11,336 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-06 14:34:11,339 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-06 14:34:11,342 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-06 14:34:11,345 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-06 14:34:11,348 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-06 14:34:11,351 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-06 14:34:11,354 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-06 14:34:11,356 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-06 14:34:11,360 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-06 14:34:11,364 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-06 14:34:11,758 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-06 14:34:11,901 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-06 14:34:11,912 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-06 14:34:11,917 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-06 14:34:11,923 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-06 14:34:11,926 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-06 14:34:11,931 autosynth [INFO] > Running synthtool
2020-07-06 14:34:11,931 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-06 14:34:11,931 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-06 14:34:11,933 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-06 14:34:12,130 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-06 14:34:12,258 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-06 14:34:12,280 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-06 14:34:12,321 autosynth [ERROR] > Synthesis failed
2020-07-06 14:34:12,321 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-06 14:34:12,327 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-06 14:34:12,332 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-06 14:34:12,539 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-06 14:34:12,539 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/2172df15-42fb-4ea2-9939-61316a413093/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
ning: git log -1 --pretty=%at ee18e132856bd1ef486249fc9fedd74b3e75f957
2020-07-07 01:34:16,665 autosynth [DEBUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-07 01:34:16,667 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-07 01:34:16,670 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-07 01:34:16,673 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-07 01:34:16,676 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-07 01:34:16,678 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-07 01:34:16,681 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-07 01:34:16,684 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-07 01:34:16,687 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-07 01:34:16,689 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-07 01:34:16,692 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-07 01:34:16,695 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-07 01:34:16,699 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-07 01:34:16,702 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-07 01:34:17,111 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-07 01:34:17,238 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-07 01:34:17,243 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-07 01:34:17,247 autosynth [DEBUG] > Running: git checkout 303271797a360f8a439203413f13a160f2f5b3b4
Note: checking out '303271797a360f8a439203413f13a160f2f5b3b4'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 3032717 chore: update Py2 support msg to reflect passage of time (#652)
2020-07-07 01:34:17,252 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-115
2020-07-07 01:34:17,255 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-115
Switched to branch 'autosynth-synthtool-115'
2020-07-07 01:34:17,260 autosynth [INFO] > Running synthtool
2020-07-07 01:34:17,260 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-07 01:34:17,260 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/115/sponge_log.log
2020-07-07 01:34:17,262 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-07 01:34:17,479 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-115
nothing to commit, working tree clean
2020-07-07 01:34:17,613 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-07 01:34:17,635 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-07 01:34:17,670 autosynth [ERROR] > Synthesis failed
2020-07-07 01:34:17,670 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-07 01:34:17,676 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-07 01:34:17,681 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-07 01:34:18,093 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-07 01:34:18,093 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/f1c80f67-23c0-4556-8f0a-ae1347d5a9f3/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
BUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-07 14:34:22,319 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-07 14:34:22,322 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-07 14:34:22,326 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-07 14:34:22,329 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-07 14:34:22,332 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-07 14:34:22,336 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-07 14:34:22,339 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-07 14:34:22,342 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-07 14:34:22,345 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-07 14:34:22,349 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-07 14:34:22,352 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-07 14:34:22,356 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-07 14:34:22,360 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-07 14:34:22,364 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-07 14:34:22,620 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-07 14:34:22,766 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-07 14:34:22,777 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-07 14:34:22,783 autosynth [DEBUG] > Running: git checkout d53e4b70cf091cba04362c2fac3cda0546121641
Note: checking out 'd53e4b70cf091cba04362c2fac3cda0546121641'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at d53e4b7 fix: set idle timeout for bazel to 240 seconds (#653)
2020-07-07 14:34:22,789 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-116
2020-07-07 14:34:22,793 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-116
Switched to branch 'autosynth-synthtool-116'
2020-07-07 14:34:22,799 autosynth [INFO] > Running synthtool
2020-07-07 14:34:22,799 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-07 14:34:22,799 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/116/sponge_log.log
2020-07-07 14:34:22,801 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-07 14:34:23,017 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-116
nothing to commit, working tree clean
2020-07-07 14:34:23,154 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-07 14:34:23,177 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-07 14:34:23,235 autosynth [ERROR] > Synthesis failed
2020-07-07 14:34:23,236 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-07 14:34:23,242 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-07 14:34:23,247 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-07 14:34:23,644 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-07 14:34:23,644 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/58020715-0b4b-44c4-95c1-cbec2e342b42/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
BUG] > Running: git log -1 --pretty=%at 483f18635c3af10031435535268abe001ad69e7c
2020-07-08 01:34:23,337 autosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-08 01:34:23,340 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-08 01:34:23,343 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-08 01:34:23,346 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-08 01:34:23,349 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-08 01:34:23,352 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-08 01:34:23,355 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-08 01:34:23,358 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-08 01:34:23,361 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-08 01:34:23,364 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-08 01:34:23,367 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-08 01:34:23,370 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-08 01:34:23,374 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-08 01:34:23,378 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-08 01:34:23,639 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-08 01:34:23,787 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-08 01:34:23,797 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-08 01:34:23,802 autosynth [DEBUG] > Running: git checkout d53e4b70cf091cba04362c2fac3cda0546121641
Note: checking out 'd53e4b70cf091cba04362c2fac3cda0546121641'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at d53e4b7 fix: set idle timeout for bazel to 240 seconds (#653)
2020-07-08 01:34:23,807 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-116
2020-07-08 01:34:23,810 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-116
Switched to branch 'autosynth-synthtool-116'
2020-07-08 01:34:23,815 autosynth [INFO] > Running synthtool
2020-07-08 01:34:23,815 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-08 01:34:23,815 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/116/sponge_log.log
2020-07-08 01:34:23,817 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-08 01:34:24,034 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-116
nothing to commit, working tree clean
2020-07-08 01:34:24,177 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-08 01:34:24,201 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-08 01:34:24,247 autosynth [ERROR] > Synthesis failed
2020-07-08 01:34:24,247 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-08 01:34:24,253 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-08 01:34:24,259 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-08 01:34:24,490 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-08 01:34:24,490 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/a58f9951-d63f-495b-a949-f8d02a497036/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
utosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-08 14:36:44,995 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-08 14:36:44,999 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-08 14:36:45,002 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-08 14:36:45,005 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-08 14:36:45,009 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-08 14:36:45,012 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-08 14:36:45,016 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-08 14:36:45,019 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-08 14:36:45,022 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-08 14:36:45,026 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-08 14:36:45,029 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-08 14:36:45,032 autosynth [DEBUG] > Running: git log -1 --pretty=%at 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
2020-07-08 14:36:45,036 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-08 14:36:45,040 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-08 14:36:45,416 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-08 14:36:45,541 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-08 14:36:45,553 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-08 14:36:45,558 autosynth [DEBUG] > Running: git checkout 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
Note: checking out '799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 799d8e6 fix: typeo in nodejs .gitattribute (#655)
2020-07-08 14:36:45,564 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-117
2020-07-08 14:36:45,567 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-117
Switched to branch 'autosynth-synthtool-117'
2020-07-08 14:36:45,572 autosynth [INFO] > Running synthtool
2020-07-08 14:36:45,572 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-08 14:36:45,572 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/117/sponge_log.log
2020-07-08 14:36:45,574 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-08 14:36:45,786 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-117
nothing to commit, working tree clean
2020-07-08 14:36:45,926 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-08 14:36:45,950 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-08 14:36:46,003 autosynth [ERROR] > Synthesis failed
2020-07-08 14:36:46,004 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-08 14:36:46,009 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-08 14:36:46,015 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-08 14:36:46,197 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-08 14:36:46,197 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/111bd893-9b98-4365-a5b5-e42fe9101d49/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
utosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-09 01:34:20,222 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-09 01:34:20,225 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-09 01:34:20,228 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-09 01:34:20,231 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-09 01:34:20,234 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-09 01:34:20,237 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-09 01:34:20,240 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-09 01:34:20,243 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-09 01:34:20,245 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-09 01:34:20,248 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-09 01:34:20,251 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-09 01:34:20,254 autosynth [DEBUG] > Running: git log -1 --pretty=%at 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
2020-07-09 01:34:20,257 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-09 01:34:20,261 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-09 01:34:20,697 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-09 01:34:20,841 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-09 01:34:20,852 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-09 01:34:20,857 autosynth [DEBUG] > Running: git checkout 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
Note: checking out '799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 799d8e6 fix: typeo in nodejs .gitattribute (#655)
2020-07-09 01:34:20,861 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-117
2020-07-09 01:34:20,864 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-117
Switched to branch 'autosynth-synthtool-117'
2020-07-09 01:34:20,869 autosynth [INFO] > Running synthtool
2020-07-09 01:34:20,869 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-09 01:34:20,869 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/117/sponge_log.log
2020-07-09 01:34:20,871 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-09 01:34:21,088 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-117
nothing to commit, working tree clean
2020-07-09 01:34:21,225 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-09 01:34:21,247 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-09 01:34:21,281 autosynth [ERROR] > Synthesis failed
2020-07-09 01:34:21,282 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-09 01:34:21,287 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-09 01:34:21,292 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-09 01:34:21,510 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-09 01:34:21,510 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/df846015-07d7-4030-b0f2-61e361036c5c/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
utosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-09 14:33:59,441 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-09 14:33:59,444 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-09 14:33:59,446 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-09 14:33:59,449 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-09 14:33:59,452 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-09 14:33:59,454 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-09 14:33:59,457 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-09 14:33:59,459 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-09 14:33:59,462 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-09 14:33:59,465 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-09 14:33:59,467 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-09 14:33:59,470 autosynth [DEBUG] > Running: git log -1 --pretty=%at 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
2020-07-09 14:33:59,473 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-09 14:33:59,477 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-09 14:33:59,832 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-09 14:33:59,954 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-09 14:33:59,963 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-09 14:33:59,968 autosynth [DEBUG] > Running: git checkout 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
Note: checking out '799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 799d8e6 fix: typeo in nodejs .gitattribute (#655)
2020-07-09 14:33:59,972 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-117
2020-07-09 14:33:59,975 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-117
Switched to branch 'autosynth-synthtool-117'
2020-07-09 14:33:59,979 autosynth [INFO] > Running synthtool
2020-07-09 14:33:59,979 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-09 14:33:59,979 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/117/sponge_log.log
2020-07-09 14:33:59,981 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-09 14:34:00,165 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-117
nothing to commit, working tree clean
2020-07-09 14:34:00,281 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-09 14:34:00,300 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-09 14:34:00,331 autosynth [ERROR] > Synthesis failed
2020-07-09 14:34:00,331 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-09 14:34:00,336 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-09 14:34:00,341 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-09 14:34:00,611 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-09 14:34:00,612 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/e9c50d75-a1ba-4ad3-a81e-eb3c5e7845ef/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
utosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-10 01:34:16,462 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-10 01:34:16,465 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-10 01:34:16,468 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-10 01:34:16,471 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-10 01:34:16,474 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-10 01:34:16,477 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-10 01:34:16,480 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-10 01:34:16,483 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-10 01:34:16,486 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-10 01:34:16,489 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-10 01:34:16,492 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-10 01:34:16,495 autosynth [DEBUG] > Running: git log -1 --pretty=%at 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
2020-07-10 01:34:16,499 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-10 01:34:16,503 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-10 01:34:16,749 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-10 01:34:16,900 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-10 01:34:16,909 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-10 01:34:16,915 autosynth [DEBUG] > Running: git checkout 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
Note: checking out '799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 799d8e6 fix: typeo in nodejs .gitattribute (#655)
2020-07-10 01:34:16,920 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-117
2020-07-10 01:34:16,924 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-117
Switched to branch 'autosynth-synthtool-117'
2020-07-10 01:34:16,928 autosynth [INFO] > Running synthtool
2020-07-10 01:34:16,928 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-10 01:34:16,929 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/117/sponge_log.log
2020-07-10 01:34:16,930 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-10 01:34:17,127 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-117
nothing to commit, working tree clean
2020-07-10 01:34:17,251 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-10 01:34:17,272 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-10 01:34:17,313 autosynth [ERROR] > Synthesis failed
2020-07-10 01:34:17,313 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-10 01:34:17,319 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-10 01:34:17,324 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-10 01:34:17,523 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-10 01:34:17,523 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/bcc99092-d90a-433b-82b1-e465dc37bcbb/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
utosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-10 14:34:17,567 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-10 14:34:17,570 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-10 14:34:17,573 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-10 14:34:17,576 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-10 14:34:17,580 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-10 14:34:17,583 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-10 14:34:17,586 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-10 14:34:17,589 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-10 14:34:17,592 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-10 14:34:17,596 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-10 14:34:17,599 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-10 14:34:17,602 autosynth [DEBUG] > Running: git log -1 --pretty=%at 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
2020-07-10 14:34:17,606 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-10 14:34:17,610 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-10 14:34:18,017 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-10 14:34:18,140 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-10 14:34:18,150 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-10 14:34:18,156 autosynth [DEBUG] > Running: git checkout 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
Note: checking out '799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 799d8e6 fix: typeo in nodejs .gitattribute (#655)
2020-07-10 14:34:18,162 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-117
2020-07-10 14:34:18,166 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-117
Switched to branch 'autosynth-synthtool-117'
2020-07-10 14:34:18,171 autosynth [INFO] > Running synthtool
2020-07-10 14:34:18,172 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-10 14:34:18,172 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/117/sponge_log.log
2020-07-10 14:34:18,174 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-10 14:34:18,373 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-117
nothing to commit, working tree clean
2020-07-10 14:34:18,502 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-10 14:34:18,523 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-10 14:34:18,567 autosynth [ERROR] > Synthesis failed
2020-07-10 14:34:18,567 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-10 14:34:18,573 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-10 14:34:18,578 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-10 14:34:18,763 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-10 14:34:18,764 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/25c51ebe-37c2-4262-ba95-6cb692394ebd/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
utosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-11 01:34:12,151 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-11 01:34:12,154 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-11 01:34:12,158 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-11 01:34:12,161 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-11 01:34:12,164 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-11 01:34:12,167 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-11 01:34:12,170 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-11 01:34:12,173 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-11 01:34:12,176 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-11 01:34:12,179 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-11 01:34:12,182 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-11 01:34:12,185 autosynth [DEBUG] > Running: git log -1 --pretty=%at 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
2020-07-11 01:34:12,189 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-11 01:34:12,193 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-11 01:34:12,462 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-11 01:34:12,608 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-11 01:34:12,618 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-11 01:34:12,624 autosynth [DEBUG] > Running: git checkout 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
Note: checking out '799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 799d8e6 fix: typeo in nodejs .gitattribute (#655)
2020-07-11 01:34:12,630 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-117
2020-07-11 01:34:12,634 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-117
Switched to branch 'autosynth-synthtool-117'
2020-07-11 01:34:12,639 autosynth [INFO] > Running synthtool
2020-07-11 01:34:12,639 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-11 01:34:12,639 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/117/sponge_log.log
2020-07-11 01:34:12,641 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-11 01:34:12,844 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-117
nothing to commit, working tree clean
2020-07-11 01:34:12,974 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-11 01:34:12,996 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-11 01:34:13,037 autosynth [ERROR] > Synthesis failed
2020-07-11 01:34:13,037 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-11 01:34:13,043 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-11 01:34:13,048 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-11 01:34:13,252 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-11 01:34:13,252 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/fb90c3c3-24b3-4b18-be76-f8d1dcf25c59/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
utosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-11 14:34:09,866 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-11 14:34:09,869 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-11 14:34:09,872 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-11 14:34:09,874 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-11 14:34:09,877 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-11 14:34:09,880 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-11 14:34:09,883 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-11 14:34:09,886 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-11 14:34:09,888 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-11 14:34:09,891 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-11 14:34:09,894 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-11 14:34:09,897 autosynth [DEBUG] > Running: git log -1 --pretty=%at 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
2020-07-11 14:34:09,900 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-11 14:34:09,904 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-11 14:34:10,170 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-11 14:34:10,313 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-11 14:34:10,323 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-11 14:34:10,327 autosynth [DEBUG] > Running: git checkout 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
Note: checking out '799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 799d8e6 fix: typeo in nodejs .gitattribute (#655)
2020-07-11 14:34:10,332 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-117
2020-07-11 14:34:10,335 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-117
Switched to branch 'autosynth-synthtool-117'
2020-07-11 14:34:10,340 autosynth [INFO] > Running synthtool
2020-07-11 14:34:10,340 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-11 14:34:10,340 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/117/sponge_log.log
2020-07-11 14:34:10,342 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-11 14:34:10,557 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-117
nothing to commit, working tree clean
2020-07-11 14:34:10,689 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-11 14:34:10,711 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-11 14:34:10,744 autosynth [ERROR] > Synthesis failed
2020-07-11 14:34:10,744 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-11 14:34:10,749 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-11 14:34:10,754 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-11 14:34:10,967 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-11 14:34:10,967 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/fcbc4554-5ed9-49d8-b1dd-c6899fe83fb5/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
utosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-12 01:34:05,934 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-12 01:34:05,937 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-12 01:34:05,940 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-12 01:34:05,943 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-12 01:34:05,946 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-12 01:34:05,949 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-12 01:34:05,952 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-12 01:34:05,955 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-12 01:34:05,957 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-12 01:34:05,961 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-12 01:34:05,964 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-12 01:34:05,966 autosynth [DEBUG] > Running: git log -1 --pretty=%at 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
2020-07-12 01:34:05,970 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-12 01:34:05,974 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-12 01:34:06,223 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-12 01:34:06,564 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-12 01:34:06,575 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-12 01:34:06,580 autosynth [DEBUG] > Running: git checkout 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
Note: checking out '799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 799d8e6 fix: typeo in nodejs .gitattribute (#655)
2020-07-12 01:34:06,585 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-117
2020-07-12 01:34:06,588 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-117
Switched to branch 'autosynth-synthtool-117'
2020-07-12 01:34:06,593 autosynth [INFO] > Running synthtool
2020-07-12 01:34:06,593 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-12 01:34:06,593 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/117/sponge_log.log
2020-07-12 01:34:06,595 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-12 01:34:06,789 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-117
nothing to commit, working tree clean
2020-07-12 01:34:06,912 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-12 01:34:06,933 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-12 01:34:06,976 autosynth [ERROR] > Synthesis failed
2020-07-12 01:34:06,976 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-12 01:34:06,982 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-12 01:34:06,987 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-12 01:34:07,181 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-12 01:34:07,181 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/a2c2a1dc-063e-4381-b60e-7d2c43ba83f3/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
utosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-12 14:34:05,640 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-12 14:34:05,643 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-12 14:34:05,646 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-12 14:34:05,649 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-12 14:34:05,651 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-12 14:34:05,654 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-12 14:34:05,657 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-12 14:34:05,660 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-12 14:34:05,663 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-12 14:34:05,666 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-12 14:34:05,668 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-12 14:34:05,671 autosynth [DEBUG] > Running: git log -1 --pretty=%at 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
2020-07-12 14:34:05,675 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-12 14:34:05,678 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-12 14:34:05,932 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-12 14:34:06,073 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-12 14:34:06,079 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-12 14:34:06,084 autosynth [DEBUG] > Running: git checkout 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
Note: checking out '799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 799d8e6 fix: typeo in nodejs .gitattribute (#655)
2020-07-12 14:34:06,089 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-117
2020-07-12 14:34:06,092 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-117
Switched to branch 'autosynth-synthtool-117'
2020-07-12 14:34:06,096 autosynth [INFO] > Running synthtool
2020-07-12 14:34:06,097 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-12 14:34:06,097 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/117/sponge_log.log
2020-07-12 14:34:06,098 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-12 14:34:06,296 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-117
nothing to commit, working tree clean
2020-07-12 14:34:06,421 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-12 14:34:06,442 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-12 14:34:06,477 autosynth [ERROR] > Synthesis failed
2020-07-12 14:34:06,478 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-12 14:34:06,483 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-12 14:34:06,488 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-12 14:34:06,695 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-12 14:34:06,695 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/9e4c4eb8-b380-4d55-a807-85094dca5983/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
utosynth [DEBUG] > Running: git log -1 --pretty=%at 873bd44c32e57f22d7afe082aa11c8d456ff2072
2020-07-13 01:34:21,466 autosynth [DEBUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-13 01:34:21,469 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-13 01:34:21,472 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-13 01:34:21,475 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-13 01:34:21,477 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-13 01:34:21,480 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-13 01:34:21,483 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-13 01:34:21,486 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-13 01:34:21,489 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-13 01:34:21,492 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-13 01:34:21,495 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-13 01:34:21,498 autosynth [DEBUG] > Running: git log -1 --pretty=%at 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
2020-07-13 01:34:21,502 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-13 01:34:21,506 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-13 01:34:21,748 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-13 01:34:21,934 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-13 01:34:21,945 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-13 01:34:21,950 autosynth [DEBUG] > Running: git checkout 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
Note: checking out '799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 799d8e6 fix: typeo in nodejs .gitattribute (#655)
2020-07-13 01:34:21,955 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-117
2020-07-13 01:34:21,958 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-117
Switched to branch 'autosynth-synthtool-117'
2020-07-13 01:34:21,963 autosynth [INFO] > Running synthtool
2020-07-13 01:34:21,963 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-13 01:34:21,963 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/117/sponge_log.log
2020-07-13 01:34:21,965 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-13 01:34:22,179 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-117
nothing to commit, working tree clean
2020-07-13 01:34:22,315 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-13 01:34:22,338 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-13 01:34:22,374 autosynth [ERROR] > Synthesis failed
2020-07-13 01:34:22,374 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-13 01:34:22,380 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-13 01:34:22,385 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-13 01:34:22,585 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-13 01:34:22,586 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/11cd6c7d-d98e-4335-9995-b3c8394f8e4a/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

Autosynth is still having trouble generating google-oauth-java-client. :sob:

Here's the output from running `synth.py`:

```
BUG] > Running: git log -1 --pretty=%at 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
2020-07-13 14:34:37,650 autosynth [DEBUG] > Running: git log -1 --pretty=%at db4f15fedc76e04018e6e43ec403e804e1142158
2020-07-13 14:34:37,655 autosynth [DEBUG] > Running: git log -1 --pretty=%at cf2eff09d0f5319a4dc5cdce2b6356d85af4a798
2020-07-13 14:34:37,660 autosynth [DEBUG] > Running: git log -1 --pretty=%at 89a7a3414aeed7c5ca82e893aa60b7058144101d
2020-07-13 14:34:37,665 autosynth [DEBUG] > Running: git log -1 --pretty=%at 2a29860e484695a07aa8670f9e750bfeeb308a43
2020-07-13 14:34:37,671 autosynth [DEBUG] > Running: git log -1 --pretty=%at ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
2020-07-13 14:34:37,676 autosynth [DEBUG] > Running: git log -1 --pretty=%at 652d446edabb0ea07de0ce542c6b37ab7dad3a19
2020-07-13 14:34:37,681 autosynth [DEBUG] > Running: git log -1 --pretty=%at d79dc99519468f365c51210436ad277757d12844
2020-07-13 14:34:37,686 autosynth [DEBUG] > Running: git log -1 --pretty=%at dc9caca650c77b7039e2bbc3339ffb34ae78e5b7
2020-07-13 14:34:37,690 autosynth [DEBUG] > Running: git log -1 --pretty=%at 303271797a360f8a439203413f13a160f2f5b3b4
2020-07-13 14:34:37,695 autosynth [DEBUG] > Running: git log -1 --pretty=%at d53e4b70cf091cba04362c2fac3cda0546121641
2020-07-13 14:34:37,700 autosynth [DEBUG] > Running: git log -1 --pretty=%at 799d8e6522c1ef7cb55a70d9ea0b15e045c3d00b
2020-07-13 14:34:37,704 autosynth [DEBUG] > Running: git log -1 --pretty=%at d82deccf657a66e31bd5da9efdb96c6fa322fc7e
2020-07-13 14:34:37,710 autosynth [DEBUG] > Running: git branch -f autosynth-self
2020-07-13 14:34:37,715 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool
2020-07-13 14:34:37,963 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/474
2020-07-13 14:34:38,111 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-13 14:34:38,123 autosynth [DEBUG] > Running: git checkout badd8974ec47e7f16a250110c2b8cf944ff28e14
Note: checking out 'badd8974ec47e7f16a250110c2b8cf944ff28e14'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-13 14:34:38,131 autosynth [DEBUG] > Running: git checkout d82deccf657a66e31bd5da9efdb96c6fa322fc7e
Note: checking out 'd82deccf657a66e31bd5da9efdb96c6fa322fc7e'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at d82decc build: missing closing paren in publish script (#659)
2020-07-13 14:34:38,138 autosynth [DEBUG] > Running: git branch -f autosynth-synthtool-118
2020-07-13 14:34:38,143 autosynth [DEBUG] > Running: git checkout autosynth-synthtool-118
Switched to branch 'autosynth-synthtool-118'
2020-07-13 14:34:38,149 autosynth [INFO] > Running synthtool
2020-07-13 14:34:38,149 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-07-13 14:34:38,150 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-oauth-java-client/synthtool/118/sponge_log.log
2020-07-13 14:34:38,153 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-07-13 14:34:38,403 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py.
On branch autosynth-synthtool-118
nothing to commit, working tree clean
2020-07-13 14:34:38,561 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-07-13 14:34:38,588 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-oauth-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-07-13 14:34:38,653 autosynth [ERROR] > Synthesis failed
2020-07-13 14:34:38,653 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at badd897 chore(deps): update project.http.version to v1.36.0 (#485)
2020-07-13 14:34:38,661 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-07-13 14:34:38,669 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-07-13 14:34:38,867 autosynth [INFO] > PR already exists: https://github.com/googleapis/google-oauth-java-client/pull/448
2020-07-13 14:34:38,867 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/bbb9fc42-c3ea-45c7-a24b-c571860301d6/targets/github%2Fsynthtool;config=default/tests;query=google-oauth-java-client;failed=false).

--- a/.github/CODEOWNERS
+++ b/.github/CODEOWNERS
@@ -1,1 +1,11 @@
-* @googleapis/yoshi-java
+# Code owners file.
+# This file controls who is tagged for review for any given pull request.
+
+# For syntax help see:
+# https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax
+
+# The @googleapis/yoshi-java is the default owner for changes in this repo
+**/*.java               @googleapis/yoshi-java
+
+# The java-samples-reviewers team is the default owner for samples changes
+samples/**/*.java       @googleapis/java-samples-reviewers

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -1,0 +1,79 @@
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+name: ci
+jobs:
+  units:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [7, 8, 11]
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v1
+      with:
+        java-version: ${{matrix.java}}
+    - run: java -version
+    - run: .kokoro/build.sh
+      env:
+        JOB_TYPE: test
+    - name: coverage
+      uses: codecov/codecov-action@v1
+      with:
+        name: actions ${{matrix.java}}
+  windows:
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 8
+    - run: java -version
+    - run: .kokoro/build.bat
+      env:
+        JOB_TYPE: test
+  dependencies:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [8, 11]
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v1
+      with:
+        java-version: ${{matrix.java}}
+    - run: java -version
+    - run: .kokoro/dependencies.sh
+  linkage-monitor:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 8
+    - run: java -version
+    - run: .kokoro/linkage-monitor.sh
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 8
+    - run: java -version
+    - run: .kokoro/build.sh
+      env:
+        JOB_TYPE: lint
+  clirr:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 8
+    - run: java -version
+    - run: .kokoro/build.sh
+      env:
+        JOB_TYPE: clirr

--- a/.kokoro/build.bat
+++ b/.kokoro/build.bat
@@ -1,3 +1,3 @@
 :: See documentation in type-shell-output.bat
 
-"C:\Program Files\Git\bin\bash.exe" github/google-oauth-java-client/.kokoro/build.sh
+"C:\Program Files\Git\bin\bash.exe" %~dp0build.sh

--- a/.kokoro/build.sh
+++ b/.kokoro/build.sh
@@ -39,7 +39,7 @@ retry_with_backoff 3 10 \
 
 # if GOOGLE_APPLICATION_CREDIENTIALS is specified as a relative path prepend Kokoro root directory onto it
 if [[ ! -z "${GOOGLE_APPLICATION_CREDENTIALS}" && "${GOOGLE_APPLICATION_CREDENTIALS}" != /* ]]; then
-    export GOOGLE_APPLICATION_CREDENTIALS=$(realpath ${KOKORO_ROOT}/src/${GOOGLE_APPLICATION_CREDENTIALS})
+    export GOOGLE_APPLICATION_CREDENTIALS=$(realpath ${KOKORO_GFILE_DIR}/${GOOGLE_APPLICATION_CREDENTIALS})
 fi
 
 RETURN_CODE=0
@@ -51,9 +51,7 @@ test)
     RETURN_CODE=$?
     ;;
 lint)
-    mvn \
-      -Penable-samples \
-      com.coveo:fmt-maven-plugin:check
+    mvn com.coveo:fmt-maven-plugin:check
     RETURN_CODE=$?
     ;;
 javadoc)

--- a/.kokoro/common.sh
+++ b/.kokoro/common.sh
@@ -52,3 +52,8 @@ function retry_with_backoff {
 
     return $exit_code
 }
+
+## Helper functionss
+function now() { date +"%Y-%m-%d %H:%M:%S" | tr -d '\n'; }
+function msg() { println "$*" >&2; }
+function println() { printf '%s\n' "$(now) $*"; }

--- a/.kokoro/dependencies.sh
+++ b/.kokoro/dependencies.sh
@@ -36,3 +36,53 @@ retry_with_backoff 3 10 \
     -Dclirr.skip=true
 
 mvn -B dependency:analyze -DfailOnWarning=true
+
+echo "****************** DEPENDENCY LIST COMPLETENESS CHECK *******************"
+## Run dependency list completeness check
+function completenessCheck() {
+  # Output dep list with compile scope generated using the original pom
+  # Running mvn dependency:list on Java versions that support modules will also include the module of the dependency.
+  # This is stripped from the output as it is not present in the flattened pom.
+  msg "Generating dependency list using original pom..."
+  mvn dependency:list -f pom.xml -Dsort=true | grep '\[INFO]    .*:.*:.*:.*:.*' | sed -e s/\\s--\\smodule.*// | grep -v ':test$' >.org-list.txt
+
+  # Output dep list generated using the flattened pom (test scope deps are ommitted)
+  msg "Generating dependency list using flattened pom..."
+  mvn dependency:list -f .flattened-pom.xml -Dsort=true | grep '\[INFO]    .*:.*:.*:.*:.*' >.new-list.txt
+
+  # Compare two dependency lists
+  msg "Comparing dependency lists..."
+  diff .org-list.txt .new-list.txt >.diff.txt
+  if [[ $? == 0 ]]
+    then
+      msg "Success. No diff!"
+  else
+    msg "Diff found. See below: "
+    msg "You can also check .diff.txt file located in $1."
+    cat .diff.txt
+    return 1
+  fi
+}
+
+# Allow failures to continue running the script
+set +e
+
+error_count=0
+for path in $(find -name ".flattened-pom.xml")
+do
+  # Check flattened pom in each dir that contains it for completeness
+  dir=$(dirname "$path")
+  pushd "$dir"
+  completenessCheck "$dir"
+  error_count=$(($error_count + $?))
+  popd
+done
+
+if [[ $error_count == 0 ]]
+then
+  msg "All checks passed."
+  exit 0
+else
+  msg "Errors found. See log statements above."
+  exit 1
+fi

--- a/.kokoro/nightly/integration.cfg
+++ b/.kokoro/nightly/integration.cfg
@@ -10,9 +10,14 @@ env_vars: {
   key: "JOB_TYPE"
   value: "integration"
 }
-
+# TODO: remove this after we've migrated all tests and scripts
 env_vars: {
   key: "GCLOUD_PROJECT"
+  value: "gcloud-devel"
+}
+
+env_vars: {
+  key: "GOOGLE_CLOUD_PROJECT"
   value: "gcloud-devel"
 }
 
@@ -22,15 +27,11 @@ env_vars: {
 }
 
 env_vars: {
-    key: "GOOGLE_APPLICATION_CREDENTIALS"
-    value: "keystore/73713_java_it_service_account"
+  key: "GOOGLE_APPLICATION_CREDENTIALS"
+  value: "secret_manager/java-it-service-account"
 }
 
-before_action {
-  fetch_keystore {
-    keystore_resource {
-      keystore_config_id: 73713
-      keyname: "java_it_service_account"
-    }
-  }
+env_vars: {
+  key: "SECRET_MANAGER_KEYS"
+  value: "java-it-service-account"
 }

--- a/.kokoro/nightly/samples.cfg
+++ b/.kokoro/nightly/samples.cfg
@@ -11,26 +11,28 @@ env_vars: {
   value: "samples"
 }
 
+# TODO: remove this after we've migrated all tests and scripts
 env_vars: {
   key: "GCLOUD_PROJECT"
-  value: "gcloud-devel"
+  value: "java-docs-samples-testing"
+}
+
+env_vars: {
+  key: "GOOGLE_CLOUD_PROJECT"
+  value: "java-docs-samples-testing"
 }
 
 env_vars: {
   key: "GOOGLE_APPLICATION_CREDENTIALS"
-  value: "keystore/73713_java_it_service_account"
+  value: "secret_manager/java-docs-samples-service-account"
+}
+
+env_vars: {
+  key: "SECRET_MANAGER_KEYS"
+  value: "java-docs-samples-service-account"
 }
 
 env_vars: {
   key: "ENABLE_BUILD_COP"
   value: "true"
-}
-
-before_action {
-  fetch_keystore {
-    keystore_resource {
-      keystore_config_id: 73713
-      keyname: "java_it_service_account"
-    }
-  }
 }

--- a/.kokoro/populate-secrets.sh
+++ b/.kokoro/populate-secrets.sh
@@ -1,0 +1,43 @@
+#!/bin/bash
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function now { date +"%Y-%m-%d %H:%M:%S" | tr -d '\n' ;}
+function msg { println "$*" >&2 ;}
+function println { printf '%s\n' "$(now) $*" ;}
+
+
+# Populates requested secrets set in SECRET_MANAGER_KEYS from service account:
+# kokoro-trampoline@cloud-devrel-kokoro-resources.iam.gserviceaccount.com
+SECRET_LOCATION="${KOKORO_GFILE_DIR}/secret_manager"
+msg "Creating folder on disk for secrets: ${SECRET_LOCATION}"
+mkdir -p ${SECRET_LOCATION}
+for key in $(echo ${SECRET_MANAGER_KEYS} | sed "s/,/ /g")
+do
+  msg "Retrieving secret ${key}"
+  docker run --entrypoint=gcloud \
+    --volume=${KOKORO_GFILE_DIR}:${KOKORO_GFILE_DIR} \
+    gcr.io/google.com/cloudsdktool/cloud-sdk \
+    secrets versions access latest \
+    --project cloud-devrel-kokoro-resources \
+    --secret ${key} > \
+    "${SECRET_LOCATION}/${key}"
+  if [[ $? == 0 ]]; then
+    msg "Secret written to ${SECRET_LOCATION}/${key}"
+  else
+    msg "Error retrieving secret ${key}"
+  fi
+done

--- a/.kokoro/presubmit/integration.cfg
+++ b/.kokoro/presubmit/integration.cfg
@@ -11,21 +11,23 @@ env_vars: {
   value: "integration"
 }
 
+# TODO: remove this after we've migrated all tests and scripts
 env_vars: {
-    key: "GCLOUD_PROJECT"
-    value: "gcloud-devel"
+  key: "GCLOUD_PROJECT"
+  value: "gcloud-devel"
 }
 
 env_vars: {
-    key: "GOOGLE_APPLICATION_CREDENTIALS"
-    value: "keystore/73713_java_it_service_account"
+  key: "GOOGLE_CLOUD_PROJECT"
+  value: "gcloud-devel"
 }
 
-before_action {
-  fetch_keystore {
-    keystore_resource {
-      keystore_config_id: 73713
-      keyname: "java_it_service_account"
-    }
-  }
+env_vars: {
+  key: "GOOGLE_APPLICATION_CREDENTIALS"
+  value: "secret_manager/java-it-service-account"
+}
+
+env_vars: {
+  key: "SECRET_MANAGER_KEYS"
+  value: "java-it-service-account"
 }

--- a/.kokoro/presubmit/samples.cfg
+++ b/.kokoro/presubmit/samples.cfg
@@ -11,21 +11,23 @@ env_vars: {
   value: "samples"
 }
 
+# TODO: remove this after we've migrated all tests and scripts
 env_vars: {
-    key: "GCLOUD_PROJECT"
-    value: "gcloud-devel"
+  key: "GCLOUD_PROJECT"
+  value: "java-docs-samples-testing"
 }
 
 env_vars: {
-    key: "GOOGLE_APPLICATION_CREDENTIALS"
-    value: "keystore/73713_java_it_service_account"
+  key: "GOOGLE_CLOUD_PROJECT"
+  value: "java-docs-samples-testing"
 }
 
-before_action {
-  fetch_keystore {
-    keystore_resource {
-      keystore_config_id: 73713
-      keyname: "java_it_service_account"
-    }
-  }
+env_vars: {
+  key: "GOOGLE_APPLICATION_CREDENTIALS"
+  value: "secret_manager/java-docs-samples-service-account"
+}
+
+env_vars: {
+  key: "SECRET_MANAGER_KEYS"
+  value: "java-docs-samples-service-account"
 }

--- a/.kokoro/trampoline.sh
+++ b/.kokoro/trampoline.sh
@@ -21,4 +21,6 @@ function cleanup() {
     echo "cleanup";
 }
 trap cleanup EXIT
+
+$(dirname $0)/populate-secrets.sh # Secret Manager secrets.
 python3 "${KOKORO_GFILE_DIR}/trampoline_v1.py"

--- a/.repo-metadata.json
+++ b/.repo-metadata.json
@@ -6,5 +6,6 @@
   "language": "java",
   "repo": "googleapis/google-oauth-java-client",
   "repo_short": "google-oauth-java-client",
+  "codeowner_team": "@googleapis/yoshi-java",
   "distribution_name": "com.google.oauth-client:google-oauth-client"
 }

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -99,7 +99,16 @@ mvn -Penable-samples clean verify
     ```
 
 2. [Activate](#profile-activation) the profile.
-3. Define your samples in a normal Maven project in the `samples/` directory
+3. Define your samples in a normal Maven project in the `samples/` directory.
+
+### Code Formatting
+
+Code in this repo is formatted with
+[google-java-format](https://github.com/google/google-java-format).
+To run formatting on your project, you can run:
+```
+mvn com.coveo:fmt-maven-plugin:format
+```
 
 ### Profile Activation
 

--- a/synth.metadata
+++ b/synth.metadata
@@ -4,14 +4,14 @@
       "git": {
         "name": ".",
         "remote": "https://github.com/googleapis/google-oauth-java-client.git",
-        "sha": "c538e8fa02391f172ccd3cb007466432ae78db51"
+        "sha": "fa8c6ba33d9066ce5b14aa59f9a2c9972a8e95aa"
       }
     },
     {
       "git": {
         "name": "synthtool",
         "remote": "https://github.com/googleapis/synthtool.git",
-        "sha": "682c0c37d1054966ca662a44259e96cc7aea4413"
+        "sha": "303271797a360f8a439203413f13a160f2f5b3b4"
       }
     }
   ]

--- a/synth.py
+++ b/synth.py
@@ -12,14 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """This script is used to synthesize generated parts of this library."""
-import synthtool as s
-import synthtool.gcp as gcp
-import logging
 
-logging.basicConfig(level=logging.DEBUG)
-common_templates = gcp.CommonTemplates()
-templates = common_templates.java_library()
-s.copy(templates, excludes=[
+import synthtool.languages.java as java
+
+java.common_templates(excludes=[
     "README.md",
     "java.header",
     "checkstyle.xml",

